gsa/src/web/pages/alerts/component.js

```diff
@@ -213,6 +220,57 @@ class AlertComponent extends React.Component {
     this.handleInteraction();
   }
 
+  openContentComposerDialog() {
+    this.setState({contentComposerDialogVisible: true});
+  }
+
+  handleOpenContentComposerDialog() {
+    const {gmp} = this.props;
+    const {filterId = NO_VALUE} = this.state;
+    gmp.filter.get({id: filterId}).then(response => {
+      this.setState({
+        filterString: isDefined(response.data) ?
+          response.data.toFilterCriteriaString() :
+          undefined,
+      });
+      this.openContentComposerDialog();
+      this.handleInteraction();
+    });
+  }
+
+  closeContentComposerDialog() {
+    const {
+      method_data_composer_include_notes,
+      method_data_composer_include_overrides,
+      filter_id,
+      filter_string,
+    } = this.state;
+    this.setState({
+      includeNotes: method_data_composer_include_notes,
+      includeOverrides: method_data_composer_include_overrides,
+      filterId: filter_id,
+      filterString: filter_string,
+      contentComposerDialogVisible: false,
+    });
+  }
+
+  handleSaveComposerContent() {
+    const {
+      includeNotes,
+      includeOverrides,
+      filterId,
+      filterString,
+    } = this.state;
+    this.setState({
+      filterId,
+      filter_id: filterId,
+      filter_string: filterString,
+      method_data_composer_include_notes: includeNotes,
+      method_data_composer_include_overrides: includeOverrides,
```

Comment:
These four lines seem to be strange to me. Why putting something in the state from the state?

Reply:
Same for **closeContentComposerDialog** method

Reply:
The values are not always the same. This duplication was needed to fulfill user expectations about closing the composer without saving and so on.